### ISSUE TITLE
Shellcheck's `source` directive for SC1090 should support either $HOME or ~tilde-expansion

### ISSUE THREAD

**Issue by ELLIOTTCABLE** [koalaman/shellcheck#2914]
#### For bugs
- Rule Id (if any, e.g. SC1000): `SC1090`
- My shellcheck version (`shellcheck --version` or "online"): `0.9.0`
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
mise_plugin_path="$HOME/.zsh/modules/wintermi/zsh-mise/zsh-mise.plugin.zsh"
if [ -f "$mise_plugin_path" ]; then
   # shellcheck source=~/.zsh/modules/wintermi/zsh-mise/zsh-mise.plugin.zsh
   . "$mise_plugin_path"

   debug_path_dump 'after zsh-mise.plugin.zsh'
fi
```

#### Here's what shellcheck currently says:
```
In Dotfiles/shenv line 267:
      . "$mise_plugin_path"
        ^-----------------^ SC1090 (warning): ShellCheck can't follow non-constant source. Use a directive to specify location.
```

----

This is related to #507, but smaller in scope: instead of supporting _analysis_ of such paths in the source, I'd like to at least see them handled _in explicit source directives_, such as the above.

At the moment, if I use a tilde in the directive (`source=~/.zsh/modules/wintermi/zsh-mise/zsh-mise.plugin.zsh`), I get SC1090, as if it didn't try to resolve it at all; and if I use an environment-variable (`source=$HOME/.zsh/modules/wintermi/zsh-mise/zsh-mise.plugin.zsh`), I get told it can't be found:

```
In Dotfiles/shenv line 267:
      . "$mise_plugin_path"
        ^-----------------^ SC1091 (info): Not following: $HOME/.zsh/modules/wintermi/zsh-mise/zsh-mise.plugin.zsh: openBinaryFile: does not exist (No such file or directory)
```

**Comment by brother** [comment]
See also #2710

**Comment by ELLIOTTCABLE** [comment]
Oh my god, and I even commented on it last fall. jfc. Closing, it's a duplicate.
